### PR TITLE
feat: tweak loops durations

### LINKED ISSUE #55: Tweak loops and wait durations
Loop durations shouldn't be based on arbitrary numbers, but based on multipliers of known "constants" (or approximate constants).

Constants we can use:
- Ethereum average block time
- Cosmos average block time

These values are hardcoded into the Peggy parameters in the genesis.

Loops:

### Eth Oracle

Retrieves events from Ethereum and sends them over to Cosmos. This one should be run slower than the average Eth block time. I suggest 2x or 3x. Edit: I'll put this a bit higher, 5x for now to reduce the number of calls to the geth node. If it's to slow we can just modify it again.

### Batch Requester

Takes whatever is in the outgoing tx pool and creates a batch; the batch will only be created if it's more profitable than the one that's already waiting in the batch queue. This should be the slowest loop, given that all validators will be calling it and the outgoing TX pool is one for the entire network. It should be a multiplier of the Cosmos average block time because that's were TX are created. I suggest we start with a value like 60x (5m) and see how it goes.
I assume we'll endup using a much higher value like 120x (10m) or 240x (20m). _I don't know how to estimate with a technical method but when you have 200 (unsynchronized) validators calling this endpoint, the amount of calls this endpoint will receive is much higher than the individual amount of times._

### Eth Signer

This gets any unsigned batches and signs them. This should be based on the cosmos block time, we should allow at least 1 block to pass before re-checking. I suggest 2x.

### Relayer

For this one we want to allow at least 1 Eth block to pass between calls.
Faster checking means a validator could be more competitive while trying to be the first to detect the contract state changes and sending new batches; but that would be at the expense of possibly more invalid TXs and more RPC calls to the Ethereum node and the Cosmos node.

## Proposal

Given that the only loop that could provide with some competitive advantage is the Relayer loop we should consider adding a flag to modify this one with a default value of 3x. Could be something like: `--relayer-loop-multiplier=1.2`

The other loops can be hardcoded as multiples of the base times.

**EDIT:**
We need to have `--requester-loop-multiplier` to make e2e tests go faster. A sensible default for a network would be 60 blocks (5m) but for a e2e test we want this to go as fast as possible.
On production, this flag won't have much use, although we might want to adjust it from time to time according to the number of TXs.
Low volume of TXs: means fewer TXs in the outgoing pool; we might want to make this value larger so we don't create too many small batches.
High volume of TXs: means more TXs in the outgoing pool; we might want to make this value smaller so we don't let too many TXs pile up in the outgoing pool.

----

## For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [x] Appropriate contributors tagged
- [x] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by facundomedica**
This PR closes #55.

Now all loops are controlled by multipliers of the corresponding block times (ETH or Cosmos). Block times are taken from the Peggy params.

Adds 2 flags:
`--relayer-loop-multiplier` controls how often the relay loop runs (in ETH block time). At a block time of 15s if we set this value to 0.5 then the relayer loop will run every 7.5s. The default value is 3.0 (45s).
`--requester-loop-multiplier` controls how often a validator requests a new batch (in Cosmos block time). By default it's set to 60.0, which at 5s per block equals to 5m.

Removes `--orch-loop-duration`

### CODE REVIEW

File: cmd/peggo/orchestrator.go [hunk 1]
```diff
@@ -219,11 +217,14 @@ func getOrchestratorCmd() *cobra.Command {
 
 	cmd.Flags().Bool(flagRelayValsets, false, "Relay validator set updates to Ethereum")
 	cmd.Flags().Bool(flagRelayBatches, false, "Relay transaction batches to Ethereum")
-	cmd.Flags().Duration(flagOrchLoopDuration, 20*time.Second, "Duration between orchestrator loops")
 	cmd.Flags().Int64(flagEthBlocksPerLoop, 40, "Number of Ethereum blocks to process per orchestrator loop")
 	cmd.Flags().String(flagCoinGeckoAPI, "https://api.coingecko.com/api/v3", "Specify the coingecko API endpoint")
 	cmd.Flags().Duration(flagEthPendingTXWait, 20*time.Minute, "Time for a pending tx to be considered stale")
 	cmd.Flags().String(flagEthAlchemyWS, "", "Specify the Alchemy websocket endpoint")
+	cmd.Flags().Float64(flagRelayerLoopMultiplier, 3.0, "Multiplier for the relayer loop duration (in ETH blocks)")
+	// nolint: lll
```
Comment (alexanderbez) [inline cmd/peggo/orchestrator.go:225]: nit: We can just apply this to the entire file at the top

File: orchestrator/main_loops.go [hunk 2]
```diff
@@ -222,12 +236,16 @@ func (p *peggyOrchestrator) EthSignerMainLoop(ctx context.Context) (err error) {
...
-	// We now send a batch request without checking for profitability, that'll be
-	// done during the relayer loop.
-	//
-	// Ref: https://github.com/umee-network/peggo/issues/55
-	return loops.RunLoop(ctx, p.logger, p.loopsDuration, func() error {
+	// Run every approximately 60 Cosmos blocks (around 5m) to allow time to receive new transactions.
+	// Running this faster will cause a lot of small batches and lots of messages going around the network.
+	// We need to remember that this call is going to be made by all the validators.
+	// This loop is configurable so it can be adjusted for E2E tests.
+	loopDuration := time.Duration(
```
Comment (alexanderbez) [inline orchestrator/main_loops.go:243]: Why not pass this variable into the `peggyOrchestrator` constructor like you do for `relayerLoopDuration`?

Comment (facundomedica) [reply orchestrator/main_loops.go:243]: I had a reason for this, but now I consider it invalid lol. Fixed!

File: cmd/peggo/orchestrator.go [hunk 3]
```diff
@@ -149,12 +150,12 @@ func getOrchestratorCmd() *cobra.Command {
...
-			// peggyParams.AverageEthereumBlockTime is in milliseconds. We add a 15% extra as a buffer so txs have time
-			// to get processed.
-			//
-			// Ref: https://github.com/umee-network/peggo/issues/55
-			averageEthBlockTime := time.Duration(peggyParams.AverageEthereumBlockTime/100*115) * time.Millisecond
+			// We multiply the relayer loop multiplier by the ETH block time.
+			// peggyParams.AverageEthereumBlockTime is in milliseconds.
+			relayerLoopDuration := time.Duration(
+				float64(peggyParams.AverageEthereumBlockTime)*
+					konfig.Float64(flagRelayerLoopMultiplier)) *
```
Comment (alexanderbez) [inline cmd/peggo/orchestrator.go:157]: 1. This is hard to read. I'd opt to put the values into variables and have `relayerLoopDuration` then be a single line.
2. How does this actually work? time.Duration is a int64.

**Review by alexanderbez — CHANGES_REQUESTED** [review]

File: cmd/peggo/orchestrator.go (hunk 3 above)
Comment (facundomedica) [reply cmd/peggo/orchestrator.go:157]: 1. Ok!
```go
	ethBlockTime := float64(peggyParams.AverageEthereumBlockTime)
	relayerLoopMultiplier := konfig.Float64(flagRelayerLoopMultiplier)
	relayerLoopDuration := time.Duration(ethBlockTime*relayerLoopMultiplier) * time.Millisecond
```

2. So what's happening is that the float gets truncated when casted, so we loose the decimals. But they are milliseconds so we lose as much as 1ms when doing this, so it shouldn't be an issue

Comment (alexanderbez) [reply cmd/peggo/orchestrator.go:157]: I see. ACK

File: cmd/peggo/orchestrator.go [hunk 4]
```diff
@@ -149,12 +150,16 @@ func getOrchestratorCmd() *cobra.Command {
...
-			averageEthBlockTime := time.Duration(peggyParams.AverageEthereumBlockTime/100*115) * time.Millisecond
+			// peggyParams.AverageBlockTime and peggyParams.AverageEthereumBlockTime are in milliseconds.
+			averageCosmosBlockTime := time.Duration(peggyParams.AverageBlockTime) * time.Millisecond
+			averageEthBlockTime := time.Duration(peggyParams.AverageEthereumBlockTime) * time.Millisecond
+
+			// We multiply the relayer loop multiplier by the ETH block time.
+			// peggyParams.AverageEthereumBlockTime is in milliseconds.
+			ethBlockTimeF64 := float64(averageEthBlockTime.Milliseconds())
+			relayerLoopMultiplier := konfig.Float64(flagRelayerLoopMultiplier)
+			// Here we cast the float64 to a Duration (int64); as we are dealing with ms, we'll lose as much as 1ms.
```
Comment (alexanderbez) [inline cmd/peggo/orchestrator.go:161]:
```suggestion

			// Here we cast the float64 to a Duration (int64); as we are dealing with ms, we'll lose as much as 1ms.
```

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**